app/jobs/export_pool_xp_to_traction_job.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+# Combine the samples from a Pool XP tube into a compound sample, generate a bioscan-pool-xp-tube-to-traction message
+# and submit it to RabbitMQ so that it can forwarded to Traction by the message processor.
+ExportPoolXpToTractionJob =
+  Struct.new(:barcode) do
+    include CompoundSampleHelper
+
+    def perform
+      message_data = get_message_data(barcode)
+
+      subject_obj = configatron.amqp.schemas.subjects[:export_pool_xp_to_traction]
+      subject = subject_obj[:subject]
+      version = subject_obj[:version]
+
+      schema = get_message_schema(subject, version)
+      encoded_message = avro_encode_message(message_data, schema)
+      send_message(encoded_message, subject, version)
+    rescue StandardError => e
+      Rails.logger.error("Error exporting Pool XP tube to Traction: <#{e.message}>")
+      raise
+    end
+
+    def fetch(uri_str, limit = 10)
+      raise IOError, 'Too many HTTP redirects' if limit.zero?
+
+      response = Net::HTTP.get_response(URI.parse(uri_str))
+
+      case response
+        when Net::HTTPSuccess     then response
+        when Net::HTTPRedirection then fetch(response['location'], limit - 1)
```

Comment:
Hi Stuart, I'm just wondering why this redirection would take place? It looks like it calls the new location upto a certain limit.

Reply:
Servers don't have to serve you the content at the current URL.  They can respond with a 301 and an associated URL which indicates the URL should be updated in future requests as it's permanent.  They can respond with 302 which is when a URL may change again in future as it's temporary.  They can respond 307 which is similar to a 302, but assigned restrictions to the HTTP method being used in the request.  It's quite complicated!

In our case, RedPanda is behind nginx and nginx is performing a redirect to the RedPanda content.  The limit is to stop following a set of redirections that go on forever or in a loop.